.eslintrc.js

```diff
@@ -136,6 +136,7 @@ module.exports = {
     ...(ENABLE_REACT_COMPILER_PLUGIN ? { 'react-compiler/react-compiler': 'error' } : {}),
     // TODO move to @mui/monorepo, codebase is moving away from default exports https://github.com/mui/material-ui/issues/21862
     'import/prefer-default-export': 'off',
+    'import/no-relative-packages': 'error',
```

Comment:
> core can't enable this 🫠 https://github.com/mui/material-ui/blob/a0ffee42815b110e14107249f193b7505d1761e5/.eslintrc.js#L245-L246

This was initially added in https://github.com/mui/material-ui/pull/34642. When I run the script again, I get only a few hits: 

```sh
/Users/oliviertassinari/material-ui/apps/pigment-css-next-app/*
/Users/oliviertassinari/material-ui/apps/pigment-css-vite-app/*
/Users/oliviertassinari/material-ui/dangerFileContent.ts
  3:32  error  Relative import from another package is not allowed. Use `size-snapshot` instead of `./scripts/sizeSnapshot`                                                          import/no-relative-packages
  4:24  error  Relative import from another package is not allowed. Use `@mui-internal/api-docs-builder/utils/replaceUrl` instead of `./packages/api-docs-builder/utils/replaceUrl`  import/no-relative-packages

/Users/oliviertassinari/material-ui/docs/scripts/updateIconSynonyms.js
  7:28  error  Relative import from another package is not allowed. Use `@mui/icons-material/renameFilters/material-design-icons` instead of `../../packages/mui-icons-material/renameFilters/material-design-icons`  import/no-relative-packages

/Users/oliviertassinari/material-ui/packages/rsc-builder/buildRsc.ts
  4:28  error  Relative import from another package is not allowed. Use `@mui-internal/api-docs-builder/utils/findComponents` instead of `../api-docs-builder/utils/findComponents`  import/no-relative-packages
  5:23  error  Relative import from another package is not allowed. Use `@mui-internal/api-docs-builder/utils/findHooks` instead of `../api-docs-builder/utils/findHooks`            import/no-relative-packages
```

So I think the right move here is

1.

```suggestion
    // TODO move to @mui/monorepo
    'import/no-relative-packages': 'error',
```

2. Enabling the rule in the mono repo and opt-in out in the few places where we do this wrong.

Reply:
Which command did you run? I get

```
✖ 648 problems (648 errors, 0 warnings)
  648 errors and 0 warnings potentially fixable with the `--fix` option.
```

When changing

```diff
-'import/no-relative-packages': 'off',
+'import/no-relative-packages': 'error',
```

and running `pnpm eslint:ci`

Reply:
@JCQuintas I had the same number of errors 648. I only collapsed all the errors that seem to be duplicates.

Reply:
I tried in https://github.com/mui/material-ui/pull/44489.